zap/src/main/java/org/parosproxy/paros/view/MainMenuBar.java

```diff
@@ -322,11 +328,13 @@ private JMenuItem getMenuFileSaveAs() {
                         public void actionPerformed(java.awt.event.ActionEvent e) {
                             if (Model.getSingleton().getSession().isNewState()) {
                                 getMenuFileControl().saveAsSession();
+                                toggleSnapshotState(true);
                             } else {
                                 View.getSingleton()
                                         .showWarningDialog(
                                                 Constant.messages.getString(
                                                         "menu.file.sessionExists.error"));
+                                toggleSnapshotState(false);
```

Comment:
If it's not new state it means that we can snapshot, right?

Reply:
Yup I think I set this one backwards.

Reply:
Actually I think I had this right. If it's in new state the save happens then the button is enabled, else (there was an error, ensure it is still disabled). I guess technically I should recheck the save state after the attempted save?

Reply:
It's true that this menu shouldn't be enabled in the first place but the snapshot shouldn't be disabled either (it wasn't before these changes).

Reply:
Okay. So revert the two changes in this method? Im unclear where this is going…

Reply:
I think it just needs to change to true (since it's the same for both branches move down).